tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemEventsTests.cs

```diff
@@ -0,0 +1,840 @@
+            Assert.That(events[3].Operation, Is.EqualTo(FileOperation.Create));
+            Assert.That(events[3].Phase, Is.EqualTo(OperationPhase.After));
+        }
+    }
+    
+    [Test]
+    public void Events_Subscribe_ToSpecificOperation_ShouldOnlyReceiveThoseEvents()
+    {
+        var fileSystem = new MockFileSystem(new MockFileSystemOptions { EnableEvents = true });
+        var createEvents = new List<FileSystemOperationEventArgs>();
+        var allEvents = new List<FileSystemOperationEventArgs>();
+        
+        using (fileSystem.Events.Subscribe(FileOperation.Create, args => createEvents.Add(args)))
+        using (fileSystem.Events.Subscribe(args => allEvents.Add(args)))
+        {
+            fileSystem.File.Create(XFS.Path(@"C:\test.txt")).Dispose();
+            fileSystem.File.WriteAllText(XFS.Path(@"C:\test.txt"), "content");
+            fileSystem.File.Delete(XFS.Path(@"C:\test.txt"));
+            
+            // Create subscription should only get Create events
+            Assert.That(createEvents.Count, Is.EqualTo(2)); // Before and After
+            foreach (var e in createEvents)
+            {
+                Assert.That(e.Operation, Is.EqualTo(FileOperation.Create));
+            }
+            
+            // All subscription should get all events
+            Assert.That(allEvents.Count, Is.EqualTo(8)); // Create(4) + Write(2) + Delete(2)
+        }
+    }
+    
+    [Test]
+    public void Events_Subscribe_ToMultipleOperations_ShouldReceiveThoseEvents()
+    {
+        var fileSystem = new MockFileSystem(new MockFileSystemOptions { EnableEvents = true });
+        var modificationEvents = new List<FileSystemOperationEventArgs>();
+        
+        using (fileSystem.Events.Subscribe(
+            new[] { FileOperation.Create, FileOperation.Write, FileOperation.Delete }, 
+            args => modificationEvents.Add(args)))
+        {
+            fileSystem.File.Create(XFS.Path(@"C:\test.txt")).Dispose();
+            fileSystem.File.WriteAllText(XFS.Path(@"C:\test.txt"), "content");
+            fileSystem.File.Delete(XFS.Path(@"C:\test.txt"));
+            
+            Assert.That(modificationEvents.Count, Is.EqualTo(6));
+            var operations = modificationEvents.Select(e => e.Operation).Distinct().ToList();
+            Assert.That(operations, Contains.Item(FileOperation.Create));
+            Assert.That(operations, Contains.Item(FileOperation.Write));
+            Assert.That(operations, Contains.Item(FileOperation.Delete));
+        }
+    }
+    
+    [Test]
+    public void Events_Unsubscribe_ShouldStopReceivingEvents()
+    {
+        var fileSystem = new MockFileSystem(new MockFileSystemOptions { EnableEvents = true });
+        var events = new List<FileSystemOperationEventArgs>();
+        
+        var subscription = fileSystem.Events.Subscribe(args => events.Add(args));
+        
+        fileSystem.File.Create(XFS.Path(@"C:\test1.txt")).Dispose();
```

Comment:
Why do you always use `File.Create` (with the need to dispose the stream) in these tests?
Especially here, where the `Dispose` of the subscription is important, it makes the test unnecessary hard to read...

I think you could replace this with:
```csharp
fileSystem.File.WriteAllText(XFS.Path(@"C:\test1.txt"), "");
```